bootstrap/pkg/kfapp/coordinator/coordinator.go

```diff
@@ -280,6 +280,28 @@ func CreateKfAppCfgFile(d *kfdefsv3.KfDef) (string, error) {
 	return cfgFilePath, cfgFilePathErr
 }
 
+// nameFromAppFile infers a default name given the path to the KFDef file.
+// returns the empty string if there is a problem getting the name.
+func nameFromAppFile(appFile string) string {
+	absAppPath, err := filepath.Abs(appFile)
+
+	if err != nil {
+		log.Errorf("KfDef.Name isn't set and there was a problem inferring the name based on the path %v; error: %v\nPlease set the name explicitly in the KFDef spec.", appFile, err)
+		return ""
```

Comment:
shouldn't we pass the errors back instead of using empty string as an indication of errors?

Reply:
I think its ok to use empty string. empty string indicates we couldn't get the default value. Passing  an error back to the caller code doesn't really help; it just forces the caller code to check the error. 

The primary reason I made this a separate function was just to write a unittest. 

The caller code provides an appropriate user facing error message in the event it ends up being the empty string.